go/rpc/encrypted_client.go

```diff
@@ -177,19 +179,38 @@ func (c *EncRPCClient) Subscribe(ctx context.Context, namespace string, ch inter
 	return subscription, nil
 }
 
-func (c *EncRPCClient) createAuthenticatedLogSubscription() (*common.LogSubscription, error) {
+func (c *EncRPCClient) createAuthenticatedLogSubscription(args []interface{}) (*common.LogSubscription, error) {
 	accountSignature, err := crypto.Sign(c.Account().Hash().Bytes(), c.viewingKey.PrivateKey.ExportECDSA())
 	if err != nil {
 		return nil, fmt.Errorf("could not sign account address to authenticate subscription. Cause: %w", err)
 	}
-	logSubscription := common.LogSubscription{
+
+	logSubscription := &common.LogSubscription{
 		SubscriptionAccount: &common.SubscriptionAccount{
 			Account:   c.Account(),
 			Signature: &accountSignature,
 		},
-		// TODO - #453 - Add the incoming filters.FilterCriteria to the common.LogSubscription.
 	}
-	return &logSubscription, nil
+
+	// If there are less than two arguments, it means no filter criteria was passed.
+	if len(args) < 2 {
+		logSubscription.Filter = &filters.FilterCriteria{}
+	} else {
+		filterCriteriaJSON, err := json.Marshal(args[1])
+		if err != nil {
+			return nil, fmt.Errorf("could not marshall filter criteria to JSON. Cause: %w", err)
+		}
+
+		filterCriteria := &filters.FilterCriteria{}
+		err = json.Unmarshal(filterCriteriaJSON, &filterCriteria)
+		if err != nil {
+			return nil, fmt.Errorf("could not unmarshall filter criteria to JSON. Cause: %w", err)
```

Comment:
My favourite typo haha, marshal is such a stupid word. Double-L here and on line 201.

Reply:
Fixed